### PR TITLE
[FSDP] Add FairScale FSDP adoptions logging

### PR DESCRIPTION
**Pull request by zhaojuanmao**
Add logging to track FairScale FSDP adoptions internally

### CODE REVIEW

**Comment by anj-s** [comment]
Thanks for adding this! Can you share how we can view these metrics?

**Review by anj-s — APPROVED** [review]

File: fairscale/nn/data_parallel/fully_sharded_data_parallel.py [hunk 1]
```diff
@@ -321,6 +321,7 @@ def __init__(
         offload_config: Optional[OffloadConfig] = None,
         state_dict_on_rank_0_only: bool = False,
     ):
+        torch._C._log_api_usage_once("fairscale.fsdp")
```
Comment (min-xu-ai) [inline fairscale/nn/data_parallel/fully_sharded_data_parallel.py:324]: Check torch version before calling this?

Comment (zhaojuanmao) [reply fairscale/nn/data_parallel/fully_sharded_data_parallel.py]: Thank you @min-xu-ai , I will do try-except and only logging if _C module is available

**Comment by zhaojuanmao** [comment]
> Thanks for adding this! Can you share how we can view these metrics?

Sure! we logged the metrics to internal scuba table, we will share the results once logging is done

**Pull request merged**